src/artm/regularizer/decorrelator_phi.cc

```diff
@@ -20,11 +21,19 @@ bool DecorrelatorPhi::RegularizePhi(const ::artm::core::PhiMatrix& p_wt,
   const int topic_size = p_wt.topic_size();
   const int token_size = p_wt.token_size();
 
-  std::vector<bool> topics_to_regularize;
-  if (config_.topic_name().size() == 0)
-    topics_to_regularize.assign(topic_size, true);
-  else
-    topics_to_regularize = core::is_member(p_wt.topic_name(), config_.topic_name());
+  std::unordered_map<std::string, int> topics_to_regularize;
```

Comment:
These changes in decorrelator_phi.cc look OK. Most importantly regularizer should work as before if new fields (first_topic_name, second_topic_name, value) are empty --- which to me seems to be the case (plus this should be already covered with unit-tests).